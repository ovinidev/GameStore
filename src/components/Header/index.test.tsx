import { render } from "@testing-library/react-native";
import { Header } from "./index";

jest.mock("@hooks/useCart", () => ({
	useCart: jest.fn().mockReturnValue({
		data: {
			carts: [],
		},
	}),
}));

jest.mock("expo-router", () => ({
	useRouter: jest.fn().mockReturnValue({
		push: jest.fn(),
	}),
	Stack: {
		Screen: jest
			.fn()
			.mockImplementation(({ options }) => <>{options.headerTitle?.()}</>),
	},
}));

describe("Header", () => {
	it("should be able to match the snapshot", () => {
		const { toJSON } = render(<Header />);

		expect(toJSON()).toMatchSnapshot();
	});
});
