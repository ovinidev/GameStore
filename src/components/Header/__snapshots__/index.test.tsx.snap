// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be able to match the snapshot 1`] = `
<Image
  resizeMode="contain"
  source="test-file-stub"
/>
`;
